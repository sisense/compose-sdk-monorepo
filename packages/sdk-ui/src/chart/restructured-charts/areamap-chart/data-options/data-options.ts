import {
  AreamapChartDataOptions,
  AreamapChartDataOptionsInternal,
  ChartDataOptions,
  ChartDataOptionsInternal,
} from '@/chart-data-options/types';
import {
  isMeasureColumn,
  normalizeColumn,
  normalizeMeasureColumn,
} from '@/chart-data-options/utils';
import { Attribute, Measure } from '@ethings-os/sdk-data';
import isArray from 'lodash-es/isArray.js';
import { ChartBuilder } from '../../types.js';

export const dataOptionsTranslators: ChartBuilder<'areamap'>['dataOptions'] = {
  translateDataOptionsToInternal: function (
    dataOptions: AreamapChartDataOptions,
  ): AreamapChartDataOptionsInternal {
    return {
      geo: dataOptions.geo && normalizeColumn(dataOptions.geo[0]),
      color: dataOptions.color && normalizeMeasureColumn(dataOptions.color[0]),
    };
  },

  getAttributes: function (internalDataOptions: AreamapChartDataOptionsInternal): Attribute[] {
    return [internalDataOptions.geo.column as Attribute].filter(
      (attribute) => !isMeasureColumn(attribute),
    );
  },

  getMeasures: function (internalDataOptions: AreamapChartDataOptionsInternal): Measure[] {
    const color = internalDataOptions.color?.column;
    return color && isMeasureColumn(color) ? [color as Measure] : [];
  },

  isCorrectDataOptions: function (
    dataOptions: ChartDataOptions,
  ): dataOptions is AreamapChartDataOptions {
    return 'geo' in dataOptions && isArray(dataOptions.geo);
  },

  isCorrectDataOptionsInternal: function (
    dataOptions: ChartDataOptionsInternal,
  ): dataOptions is AreamapChartDataOptionsInternal {
    return 'geo' in dataOptions && !!dataOptions.geo.column;
  },
};
