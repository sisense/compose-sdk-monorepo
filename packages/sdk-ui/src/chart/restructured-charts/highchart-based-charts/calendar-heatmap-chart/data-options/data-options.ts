import { Attribute, Measure, DateLevels } from '@sisense/sdk-data';
import {
  CalendarHeatmapChartDataOptions,
  CalendarHeatmapChartDataOptionsInternal,
} from '@/chart-data-options/types';
import { ChartDataOptions } from '@/types';
import { ChartDataOptionsInternal } from '@/chart-data-options/types';
import {
  isMeasureColumn,
  normalizeColumn,
  normalizeMeasureColumn,
} from '@/chart-data-options/utils';
import { getDefaultDateFormat } from '@/chart-options-processor/translations/axis-section';
import { isAttributeColumn } from '../../cartesians/helpers/data-options.js';

/**
 * Translates calendar heatmap chart data options from user format to internal format
 *
 * Converts user-provided data options to the internal format used by the chart engine,
 * applying default date formatting and normalizing columns.
 *
 * @param dataOptions - User-provided calendar heatmap chart data options
 * @returns Normalized internal data options
 */
export function translateCalendarHeatmapChartDataOptions(
  dataOptions: CalendarHeatmapChartDataOptions,
): CalendarHeatmapChartDataOptionsInternal {
  const { date, value } = dataOptions;

  return {
    date: normalizeColumn(date, { dateFormat: getDefaultDateFormat(DateLevels.Days) }),
    value: normalizeMeasureColumn(value),
  };
}

/**
 * Extracts attributes from calendar heatmap chart data options
 *
 * Returns an array of attributes that can be used for data queries.
 * For calendar heatmaps, this typically includes the date attribute.
 *
 * @param dataOptions - Internal calendar heatmap chart data options
 * @returns Array of attributes extracted from the data options
 */
export function getCalendarHeatmapAttributes(
  dataOptions: CalendarHeatmapChartDataOptionsInternal,
): Attribute[] {
  const column = dataOptions.date.column;
  return isAttributeColumn(column) ? [column as Attribute] : [];
}

/**
 * Extracts measures from calendar heatmap chart data options
 *
 * Returns an array of measures that can be used for data queries.
 * For calendar heatmaps, this typically includes the value measure.
 *
 * @param dataOptions - Internal calendar heatmap chart data options
 * @returns Array of measures extracted from the data options
 */
export function getCalendarHeatmapMeasures(
  dataOptions: CalendarHeatmapChartDataOptionsInternal,
): Measure[] {
  const column = dataOptions.value.column;
  return isMeasureColumn(column) ? [column as Measure] : [];
}

/**
 * Type guard to check if data options are valid for calendar heatmap charts
 *
 * Validates that the provided data options contain the required properties
 * for calendar heatmap charts (date and value).
 *
 * @param dataOptions - Data options to validate
 * @returns True if the data options are valid for calendar heatmap charts
 */
export function isCalendarHeatmapChartDataOptions(
  dataOptions: ChartDataOptions,
): dataOptions is CalendarHeatmapChartDataOptions {
  return 'date' in dataOptions && 'value' in dataOptions;
}

/**
 * Type guard to check if internal data options are valid for calendar heatmap charts
 *
 * Validates that the provided internal data options contain the required properties
 * and are properly structured for calendar heatmap charts.
 *
 * @param dataOptions - Internal data options to validate
 * @returns True if the internal data options are valid for calendar heatmap charts
 */
export function isCalendarHeatmapChartDataOptionsInternal(
  dataOptions: ChartDataOptionsInternal,
): dataOptions is CalendarHeatmapChartDataOptionsInternal {
  return (
    'date' in dataOptions &&
    'value' in dataOptions &&
    typeof dataOptions.date === 'object' &&
    typeof dataOptions.value === 'object'
  );
}
