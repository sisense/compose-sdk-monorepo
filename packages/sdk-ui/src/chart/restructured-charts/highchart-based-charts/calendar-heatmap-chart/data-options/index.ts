export * from './data-options.js';
