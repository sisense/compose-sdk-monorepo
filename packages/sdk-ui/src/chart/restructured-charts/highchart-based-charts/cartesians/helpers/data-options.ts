import isArray from 'lodash-es/isArray';
import {
  CartesianChartDataOptionsInternal,
  ChartDataOptions,
  ChartDataOptionsInternal,
  StyledColumn,
  StyledMeasureColumn,
} from '@/chart-data-options/types';
import {
  isMeasureColumn,
  isStyledColumn,
  normalizeColumn,
  normalizeMeasureColumn,
} from '@/chart-data-options/utils';
import { CartesianChartDataOptions } from '@/types';
import {
  Attribute,
  CalculatedMeasureColumn,
  Column,
  Measure,
  MeasureColumn,
} from '@ethings-os/sdk-data';

/**
 * Translates the data options for cartesian charts to internal format.
 */
export const translateCartesianChartDataOptions = (
  dataOptions: CartesianChartDataOptions,
): CartesianChartDataOptionsInternal => {
  return {
    x: dataOptions.category.map((c) => normalizeColumn(c)),
    y: dataOptions.value.map((c) => normalizeMeasureColumn(c)),
    // breakBy may be undefined. If so, default to empty array
    breakBy: dataOptions.breakBy?.map((c) => normalizeColumn(c)) || [],
    seriesToColorMap: dataOptions.seriesToColorMap,
  };
};

/**
 * Returns the attributes from the internal data options for cartesian charts.
 */
export function getCartesianAttributes(
  internalDataOptions: CartesianChartDataOptionsInternal,
): Attribute[] {
  return [
    ...internalDataOptions.x.map(getColumn),
    ...internalDataOptions.breakBy.map(getColumn),
  ].filter(isAttributeColumn) as Attribute[];
}

/**
 * Returns the measures from the internal data options for cartesian charts.
 */
export function getCartesianMeasures(
  internalDataOptions: CartesianChartDataOptionsInternal,
): Measure[] {
  return internalDataOptions.y.map(getColumn).filter(isMeasureColumn) as Measure[];
}

/**
 * Checks if the data options are correct for cartesian charts.
 */
export function isCartesianChartDataOptions(
  dataOptions: ChartDataOptions,
): dataOptions is CartesianChartDataOptions {
  return (
    'category' in dataOptions &&
    isArray(dataOptions.category) &&
    'value' in dataOptions &&
    isArray(dataOptions.value) &&
    'breakBy' in dataOptions &&
    isArray(dataOptions.breakBy)
  );
}

/**
 * Checks if the internal data options are correct for cartesian charts.
 */
export function isCartesianChartDataOptionsInternal(
  dataOptions: ChartDataOptionsInternal,
): dataOptions is CartesianChartDataOptionsInternal {
  return (
    'x' in dataOptions &&
    isArray(dataOptions.x) &&
    dataOptions.x.every(isStyledColumn) &&
    'y' in dataOptions &&
    isArray(dataOptions.y) &&
    dataOptions.y.every(isStyledColumn) &&
    'breakBy' in dataOptions &&
    isArray(dataOptions.breakBy) &&
    dataOptions.breakBy.every(isStyledColumn)
  );
}

/**
 * Returns the column from the styled column.
 */
function getColumn(
  styledColumn: StyledColumn | StyledMeasureColumn,
): MeasureColumn | CalculatedMeasureColumn | Column {
  return styledColumn.column;
}

/**
 * Checks if the column is an attribute column.
 */
export function isAttributeColumn(
  column: MeasureColumn | CalculatedMeasureColumn | Column,
): column is Column {
  return !isMeasureColumn(column);
}

/**
 * Checks if the chart should have Y2 axis.
 */
export function shouldHaveY2Axis(dataOptions: CartesianChartDataOptionsInternal): boolean {
  return dataOptions.y.some((y) => y.showOnRightAxis);
}
