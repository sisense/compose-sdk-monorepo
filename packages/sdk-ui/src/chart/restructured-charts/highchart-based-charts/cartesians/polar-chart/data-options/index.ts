import {
  getCartesianAttributes,
  getCartesianMeasures,
  isCartesianChartDataOptions,
  isCartesianChartDataOptionsInternal,
  translateCartesianChartDataOptions,
} from '../../helpers/data-options';

/**
 * Data options translators for polar charts.
 * Polar charts use the same data structure as cartesian charts.
 */
export const dataOptionsTranslators = {
  translateDataOptionsToInternal: translateCartesianChartDataOptions,
  getAttributes: getCartesianAttributes,
  getMeasures: getCartesianMeasures,
  isCorrectDataOptions: isCartesianChartDataOptions,
  isCorrectDataOptionsInternal: isCartesianChartDataOptionsInternal,
};
