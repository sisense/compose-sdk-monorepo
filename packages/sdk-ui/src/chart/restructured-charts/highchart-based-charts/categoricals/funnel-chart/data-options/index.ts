import { Attribute, Measure } from '@sisense/sdk-data';
import flow from 'lodash-es/flow';

import { ChartDataOptions, ChartDataOptionsInternal } from '@/chart-data-options/types';
import {
  withCategoryLimitation,
  withValueLimitation,
} from '@/chart-data-options/validate-data-options/validate-categorical-data-options';

import {
  getCategoricalAttributes,
  getCategoricalMeasures,
  isCategoricalChartDataOptions,
  isCategoricalChartDataOptionsInternal,
  translateCategoricalDataOptionsToInternal,
} from '../../helpers/data-options';
import { FunnelChartDataOptions, FunnelChartDataOptionsInternal } from '../types';

const MAX_CATEGORICAL_DATA_OPTIONS_LENGTHS = {
  category: 3,
  value: 1,
} as const;

/**
 * Data options translators for funnel charts.
 */
export const dataOptionsTranslators = {
  /**
   * Translates funnel chart data options to internal format.
   */
  translateDataOptionsToInternal: (
    dataOptions: FunnelChartDataOptions,
  ): FunnelChartDataOptionsInternal => {
    return translateCategoricalDataOptionsToInternal(dataOptions);
  },

  /**
   * Gets attributes from funnel chart data options.
   */
  getAttributes: (internalDataOptions: FunnelChartDataOptionsInternal): Attribute[] => {
    return getCategoricalAttributes(internalDataOptions);
  },

  /**
   * Gets measures from funnel chart data options.
   */
  getMeasures: (internalDataOptions: FunnelChartDataOptionsInternal): Measure[] => {
    return getCategoricalMeasures(internalDataOptions);
  },

  /**
   * Validates and cleans funnel chart data options.
   */
  validateAndCleanDataOptions: (dataOptions: FunnelChartDataOptions): FunnelChartDataOptions => {
    return flow(
      withCategoryLimitation(MAX_CATEGORICAL_DATA_OPTIONS_LENGTHS.category),
      dataOptions.category.length > 0
        ? withValueLimitation(MAX_CATEGORICAL_DATA_OPTIONS_LENGTHS.value)
        : (data: FunnelChartDataOptions) => data,
    )(dataOptions);
  },

  /**
   * Type guard to check if data options are funnel chart data options.
   */
  isCorrectDataOptions: (dataOptions: ChartDataOptions): dataOptions is FunnelChartDataOptions => {
    return isCategoricalChartDataOptions(dataOptions);
  },

  /**
   * Type guard to check if internal data options are funnel chart data options.
   */
  isCorrectDataOptionsInternal: (
    dataOptions: ChartDataOptionsInternal,
  ): dataOptions is FunnelChartDataOptionsInternal => {
    return isCategoricalChartDataOptionsInternal(dataOptions);
  },
};
