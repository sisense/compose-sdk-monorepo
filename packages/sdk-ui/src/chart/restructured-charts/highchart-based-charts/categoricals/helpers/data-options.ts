import {
  Attribute,
  CalculatedMeasureColumn,
  Column,
  Measure,
  MeasureColumn,
} from '@sisense/sdk-data';
import {
  ChartDataOptions,
  CategoricalChartDataOptions,
  CategoricalChartDataOptionsInternal,
  StyledMeasureColumn,
  StyledColumn,
} from '@/chart-data-options/types';
import {
  isMeasureColumn,
  isStyledColumn,
  normalizeColumn,
  normalizeMeasureColumn,
} from '@/chart-data-options/utils';

/**
 * Translates categorical chart data options to internal format.
 */
export function translateCategoricalDataOptionsToInternal(
  dataOptions: CategoricalChartDataOptions,
): CategoricalChartDataOptionsInternal {
  return {
    y: dataOptions.value.map((c) => normalizeMeasureColumn(c)),
    breakBy: dataOptions.category.map((c) => normalizeColumn(c)),
    seriesToColorMap: dataOptions.seriesToColorMap,
  };
}

/**
 * Gets attributes from categorical chart data options.
 */
export function getCategoricalAttributes(
  internalDataOptions: CategoricalChartDataOptionsInternal,
): Attribute[] {
  return [...internalDataOptions.breakBy.map(getColumn)].filter(isAttributeColumn) as Attribute[];
}

/**
 * Gets measures from categorical chart data options.
 */
export function getCategoricalMeasures(
  internalDataOptions: CategoricalChartDataOptionsInternal,
): Measure[] {
  return internalDataOptions.y.map(getColumn).filter(isMeasureColumn) as Measure[];
}

/**
 * Type guard to check if data options are categorical chart data options.
 */
export function isCategoricalChartDataOptions(
  dataOptions: ChartDataOptions,
): dataOptions is CategoricalChartDataOptions {
  return (
    'category' in dataOptions &&
    'value' in dataOptions &&
    Array.isArray((dataOptions as CategoricalChartDataOptions).category) &&
    Array.isArray((dataOptions as CategoricalChartDataOptions).value)
  );
}

/**
 * Type guard to check if internal data options are categorical chart data options.
 */
export function isCategoricalChartDataOptionsInternal(
  dataOptions: any,
): dataOptions is CategoricalChartDataOptionsInternal {
  return (
    dataOptions &&
    'y' in dataOptions &&
    Array.isArray(dataOptions.y) &&
    dataOptions.y.every(isMeasureColumn) &&
    'breakBy' in dataOptions &&
    Array.isArray(dataOptions.breakBy) &&
    dataOptions.breakBy.every(isStyledColumn)
  );
}

/**
 * Returns the column from the styled column.
 */
function getColumn(
  styledColumn: StyledColumn | StyledMeasureColumn,
): MeasureColumn | CalculatedMeasureColumn | Column {
  return styledColumn.column;
}

/**
 * Checks if the column is an attribute column.
 */
function isAttributeColumn(
  column: MeasureColumn | CalculatedMeasureColumn | Column,
): column is Column {
  return !isMeasureColumn(column);
}
