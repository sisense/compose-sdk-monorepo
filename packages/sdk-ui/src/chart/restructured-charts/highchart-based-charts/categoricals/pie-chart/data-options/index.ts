import { Attribute, Measure } from '@sisense/sdk-data';
import flow from 'lodash-es/flow';

import { ChartDataOptions, ChartDataOptionsInternal } from '@/chart-data-options/types';
import {
  withCategoryLimitation,
  withValueLimitation,
} from '@/chart-data-options/validate-data-options/validate-categorical-data-options';

import {
  getCategoricalAttributes,
  getCategoricalMeasures,
  isCategoricalChartDataOptions,
  isCategoricalChartDataOptionsInternal,
  translateCategoricalDataOptionsToInternal,
} from '../../helpers/data-options';
import { PieChartDataOptions, PieChartDataOptionsInternal } from '../types';

const MAX_CATEGORICAL_DATA_OPTIONS_LENGTHS = {
  category: 3,
  value: 1,
} as const;

/**
 * Data options translators for pie charts.
 */
export const dataOptionsTranslators = {
  /**
   * Translates pie chart data options to internal format.
   */
  translateDataOptionsToInternal: (
    dataOptions: PieChartDataOptions,
  ): PieChartDataOptionsInternal => {
    return translateCategoricalDataOptionsToInternal(dataOptions);
  },

  /**
   * Gets attributes from pie chart data options.
   */
  getAttributes: (internalDataOptions: PieChartDataOptionsInternal): Attribute[] => {
    return getCategoricalAttributes(internalDataOptions);
  },

  /**
   * Gets measures from pie chart data options.
   */
  getMeasures: (internalDataOptions: PieChartDataOptionsInternal): Measure[] => {
    return getCategoricalMeasures(internalDataOptions);
  },

  /**
   * Validates and cleans pie chart data options.
   */
  validateAndCleanDataOptions: (dataOptions: PieChartDataOptions): PieChartDataOptions => {
    return flow(
      withCategoryLimitation(MAX_CATEGORICAL_DATA_OPTIONS_LENGTHS.category),
      dataOptions.category.length > 0
        ? withValueLimitation(MAX_CATEGORICAL_DATA_OPTIONS_LENGTHS.value)
        : (data: PieChartDataOptions) => data,
    )(dataOptions);
  },

  /**
   * Type guard to check if data options are pie chart data options.
   */
  isCorrectDataOptions: (dataOptions: ChartDataOptions): dataOptions is PieChartDataOptions => {
    return isCategoricalChartDataOptions(dataOptions);
  },

  /**
   * Type guard to check if internal data options are pie chart data options.
   */
  isCorrectDataOptionsInternal: (
    dataOptions: ChartDataOptionsInternal,
  ): dataOptions is PieChartDataOptionsInternal => {
    return isCategoricalChartDataOptionsInternal(dataOptions);
  },
};
