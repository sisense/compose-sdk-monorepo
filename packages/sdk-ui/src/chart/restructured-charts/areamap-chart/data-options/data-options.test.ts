import { dataOptionsTranslators } from './data-options';
import * as DM from '@/__test-helpers__/sample-ecommerce';
import { measureFactory } from '@sisense/sdk-data';

import { AreamapChartDataOptions, AreamapChartDataOptionsInternal } from '@/index';

describe('areamap - data options translators', () => {
  describe('translateDataOptionsToInternal', () => {
    it('should return an object with geo and color properties', () => {
      const dataOptions: AreamapChartDataOptions = {
        geo: [
          {
            column: DM.Country.Country,
            isColored: false,
          },
        ],
        color: [
          {
            column: measureFactory.sum(DM.Commerce.Cost),
            color: {
              type: 'range',
              steps: 9,
              minColor: '#ffab03',
              maxColor: '#ff0000',
            },
          },
        ],
      };
      const result = dataOptionsTranslators.translateDataOptionsToInternal(dataOptions);
      expect(result).toMatchSnapshot();
    });
  });

  describe('getAttributes', () => {
    it('should return an attribute if geo.column is an attribute column', () => {
      const internalDataOptions: AreamapChartDataOptionsInternal = {
        geo: { column: DM.Commerce.AgeRange },
      };
      const result = dataOptionsTranslators.getAttributes(internalDataOptions);
      expect(result).toEqual([DM.Commerce.AgeRange]);
    });

    it('should return an empty array if geo.column is a measure column', () => {
      const internalDataOptions: AreamapChartDataOptionsInternal = {
        geo: { column: measureFactory.sum(DM.Commerce.Cost) },
      };
      const result = dataOptionsTranslators.getAttributes(internalDataOptions);
      expect(result).toEqual([]);
    });
  });

  describe('getMeasures', () => {
    it('should return the measure if color.column is a measure column', () => {
      const TotalCostMeasure = measureFactory.sum(DM.Commerce.Cost);
      const internalDataOptions: AreamapChartDataOptionsInternal = {
        geo: { column: DM.Country.Country },
        color: { column: TotalCostMeasure },
      };
      const result = dataOptionsTranslators.getMeasures(internalDataOptions);
      expect(result).toEqual([TotalCostMeasure]);
    });

    it('should return an empty array if color.column is not a measure column', () => {
      const internalDataOptions: AreamapChartDataOptionsInternal = {
        geo: { column: DM.Country.Country },
        color: { column: DM.Commerce.AgeRange },
      };
      const result = dataOptionsTranslators.getMeasures(internalDataOptions);
      expect(result).toEqual([]);
    });

    it('should return an empty array if color is undefined', () => {
      const internalDataOptions = { geo: { column: {} } } as any;
      const result = dataOptionsTranslators.getMeasures(internalDataOptions);
      expect(result).toEqual([]);
    });
  });

  describe('isCorrectDataOptions', () => {
    it('should return true if dataOptions has a geo property that is an array', () => {
      const dataOptions = { geo: [] };
      const result = dataOptionsTranslators.isCorrectDataOptions(dataOptions);
      expect(result).toBe(true);
    });

    it('should return false if geo property is missing', () => {
      const dataOptions = { notGeo: [] } as any;
      const result = dataOptionsTranslators.isCorrectDataOptions(dataOptions);
      expect(result).toBe(false);
    });

    it('should return false if geo is not an array', () => {
      const dataOptions = { geo: 'not an array' } as any;
      const result = dataOptionsTranslators.isCorrectDataOptions(dataOptions);
      expect(result).toBe(false);
    });
  });

  describe('isCorrectDataOptionsInternal', () => {
    it('should return true if internal data options has geo with a correct column', () => {
      const dataOptionsInternal: AreamapChartDataOptionsInternal = {
        geo: { column: DM.Country.Country },
      };
      const result = dataOptionsTranslators.isCorrectDataOptionsInternal(dataOptionsInternal);
      expect(result).toBe(true);
    });

    it('should return false if geo property is missing', () => {
      const dataOptionsInternal = {
        notGeo: { column: DM.Country.Country },
      } as unknown as AreamapChartDataOptionsInternal;
      const result = dataOptionsTranslators.isCorrectDataOptionsInternal(dataOptionsInternal);
      expect(result).toBe(false);
    });

    it('should return false if geo.column is falsy', () => {
      const dataOptionsInternal = {
        geo: { column: null },
      } as unknown as AreamapChartDataOptionsInternal;
      const result = dataOptionsTranslators.isCorrectDataOptionsInternal(dataOptionsInternal);
      expect(result).toBe(false);
    });
  });
});
