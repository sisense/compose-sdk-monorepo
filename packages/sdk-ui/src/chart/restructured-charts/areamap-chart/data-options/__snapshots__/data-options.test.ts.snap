// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`areamap - data options translators > translateDataOptionsToInternal > should return an object with geo and color properties 1`] = `
{
  "color": {
    "color": {
      "maxColor": "#ff0000",
      "minColor": "#ffab03",
      "steps": 9,
      "type": "range",
    },
    "column": {
      "__serializable": "DimensionalBaseMeasure",
      "aggregation": "sum",
      "attribute": {
        "__serializable": "DimensionalAttribute",
        "composeCode": "DM.Commerce.Cost",
        "dataSource": {
          "live": false,
          "title": "Sample ECommerce",
        },
        "description": "",
        "expression": "[Commerce.Cost]",
        "name": "Cost",
        "type": "numeric-attribute",
      },
      "composeCode": "measureFactory.sum(DM.Commerce.Cost)",
      "dataSource": undefined,
      "description": "",
      "name": "sum Cost",
      "sort": 0,
      "type": "basemeasure",
    },
    "enabled": true,
  },
  "geo": {
    "column": {
      "__serializable": "DimensionalAttribute",
      "composeCode": "DM.Country.Country",
      "dataSource": {
        "live": false,
        "title": "Sample ECommerce",
      },
      "description": "",
      "expression": "[Country.Country]",
      "name": "Country",
      "type": "text-attribute",
    },
    "isColored": false,
  },
}
`;
